samza-api/src/main/java/org/apache/samza/context/TaskContext.java

```diff
@@ -0,0 +1,70 @@
+package org.apache.samza.context;
+
+import java.util.Set;
+import org.apache.samza.annotation.InterfaceStability;
+import org.apache.samza.container.TaskName;
+import org.apache.samza.metrics.MetricsRegistry;
+import org.apache.samza.scheduling.Scheduler;
+import org.apache.samza.storage.kv.KeyValueStore;
+import org.apache.samza.system.SystemStreamPartition;
+import org.apache.samza.table.Table;
+
+
+/**
+ * Contains information at task granularity, provided by the Samza framework, to be used to instantiate an application
+ * at runtime.
+ * <p>
+ * Note that application-defined task-level context is accessible through {@link ApplicationDefinedTaskContext}.
+ */
+public interface TaskContext {
+  /**
+   * @return {@link TaskName} for this task
+   */
+  TaskName getTaskName();
+
+  /**
+   * Returns all of the input {@link SystemStreamPartition}s for this task. This does not include side input
+   * {@link SystemStreamPartition}s.
+   * @return all of the input {@link SystemStreamPartition}s for this task
+   */
+  Set<SystemStreamPartition> getSystemStreamPartitions();
```

Comment:
Do we want this in the new API? Should be able to get this from the JobModel.

Reply:
Will replace this with TaskModel.